images/haproxy/Dockerfile

```diff
@@ -15,7 +15,7 @@
 # This image is a haproxy image + minimal config so the container will not exit
 # while we rewrite the config at runtime and signal haproxy to reload.
 
-ARG BASE="k8s.gcr.io/build-image/debian-base:buster-v1.8.0"
+ARG BASE="k8s.gcr.io/build-image/debian-base:buster-v1.9.0"
```

Comment:
something broke with out image with this change (just built and tested it) so this won't be so simple.

Reply:
```
haproxy: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory
haproxy: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory
```

Reply:
probably a bug in `stage-binary-and-deps.sh` or something changed with the env in the base image

Reply:
@BenTheElder I will try to investigate this but I am close to long vacation next month, so may not get around to providing a fix. Happy for someone else to pick this up if this is urgent.

Reply:
no worries! this is going to be a fun one and I think we're OK in the meantime, I think we can circle back to haproxy and just bump the rest for now.

Most of this image is only used at build time anyhow.

Reply:
@BenTheElder Thanks! Should I break the PR into two then? i.e. Remove the ha-proxy into a new PR to be investigated and keep the rest here so we can merge that and get it bumped?

Reply:
yes please!

Reply:
@BenTheElder removed haproxy image update as requested!